src/lsp-server.ts

```diff
@@ -38,6 +38,44 @@ import { computeCallers, computeCallees } from './calls';
 import { IServerOptions } from './utils/configuration';
 import { TypeScriptVersion, TypeScriptVersionProvider } from './utils/versionProvider';
 import { TypeScriptAutoFixProvider } from './features/fix-all';
+import { LspClient, ProgressReporter } from './lsp-client';
+
+class ServerInitializingIndicator {
+    private _loadingProjectName?: string;
+    private _progressReporter?: ProgressReporter;
+
+    constructor(private lspClient: LspClient) {}
+
+    public reset(): void {
+        if (this._loadingProjectName) {
```

Comment:
do you like early returns to keep the indentation minimal?

Reply:
Since this is based on typescript-language-features, I wasn't really trying to rewrite everything.

Generally I prefer early returns, yes.

Reply:
In this case it would result in more lines which is another thing I prefer to avoid though :)